edk2toollib/windows/policy/firmware_policy.py

```diff
@@ -0,0 +1,641 @@
+        if Type not in self.SupportedValueTypes:
+            ErrorMessage = ('Unsupported ValueType: %x' % Type)
+            print(ErrorMessage)
+            if STRICT:
+                raise Exception(ErrorMessage)
+
+        self.vt = Type
+
+    """if offset is not specified, stream fs position is at beginning PolicyValueType struct"""
+    @classmethod
+    def FromFileStream(cls, fs: BinaryIO, fsOffset: int = None):
+        if fsOffset:
+            fs.seek(fsOffset)
+        valueType = struct.unpack('<H', fs.read(2))[0]
+        return cls(valueType)
+
+    @classmethod
+    def FromBytes(cls, b: bytes, bOffset: int = 0):
+        valueType = struct.unpack_from('<H', b, bOffset)[0]
+        return cls(valueType)
+
+    def Print(self, prefix: str = ''):
+        print('%s%s%s' % (prefix, 'ValueType: ', self.vt))
+
+    def Serialize(self, valueOut: bytearray):
+        valueOut += struct.pack(self.StructFormat, self.vt)
+
+    def Get(self):
+        return self.vt
+
+
+class PolicyString():
+    """
+    Class for storing, serializing, deserializing, & printing PolicyString Types
+    NOTE: This type is used both in the keys as SubKeyName and ValueName and it
+        can be a value.  When used as a value, a NULL may follow the string, but
+        the NULL will not be included in the string size
+
+    16-bit, little endian, size of the string in _bytes_
+    followed by a UTF-16LE string, no NULL terminator
+
+    Example of "PlatformID"
+    \x14\x00P\x00l\x00a\x00t\x00f\x00o\x00r\x00m\x00I\x00D\x00
+
+    The trailing \x00 is not a NULL, it is UTF-16LE encoding of "D"
+    """
+    StringLengthFormat = '<H'
+
+    def __init__(self, String: str = None):
+        if String:
+            self.String = String
+        else:
+            self.String = ''
+        return
+
+    @classmethod
+    def FromFileStream(cls, fs: BinaryIO, fsOffset: int = None):
+        if fsOffset:
+            fs.seek(fsOffset)
+        StringLength = struct.unpack('<H', fs.read(2))[0]
+        LocalString = bytes.decode(
+            fs.read(StringLength), encoding='utf_16_le')
+        if (len(LocalString) != (StringLength / 2)):
+            raise Exception('String length mismatch')
+        return cls(String=LocalString)
+
+    @classmethod
+    def FromBytes(cls, b: bytes, bOffset: int = 0):
+        StringLength = struct.unpack_from(cls.StringLengthFormat, b, bOffset)[0]
+        bOffset += struct.calcsize(cls.StringLengthFormat)
+        LocalString = bytes.decode(
+            b[bOffset: bOffset + StringLength], encoding='utf_16_le')
+        if (len(LocalString) != (StringLength / 2)):
+            raise Exception('String length mismatch')
+        return cls(String=LocalString)
+
+    def Print(self, prefix: str = ''):
+        print('%s%s%s' % (prefix, 'String:  ', self.String))
+
+    def Serialize(self, valueOut: bytearray):
+        b = str.encode(self.String, encoding='utf_16_le')
+        size = struct.pack('<H', len(b))
```

Comment:
Is this the length? self.StringLengthFormat ?

Reply:
You caught me again, refactored part of the code but dropped the other... will fix asap, thanks!

Reply:
Addressed in latest commit